event-sourcing/event-publisher/event-publisher-process/src/main/java/uk/gov/justice/services/eventsourcing/publishing/SequenceNumberUpdater.java

```diff
@@ -0,0 +1,87 @@
+    private static final String SELECT_PREVIOUS_SEQUENCE_NUMBER_SQL = "SELECT sequence_number FROM event_sequence WHERE next = 0 FOR UPDATE";
+    private static final String UPDATE_NEXT_COLUMN_IN_PREVIOUS_ROW_SQL = "UPDATE event_sequence SET next = ? WHERE next = 0";
+    private static final String INSERT_NEW_ROW_SQL = "INSERT INTO event_sequence (event_id, sequence_number, previous, next) VALUES (?, ?, ?, ?)";
+
+    @Inject
+    SubscriptionDataSourceProvider subscriptionDataSourceProvider;
+
+    @Transactional(MANDATORY)
+    public long update(final UUID eventId) throws SQLException {
+
+        try (final Connection connection = subscriptionDataSourceProvider.getEventStoreDataSource().getConnection()) {
+
+            final long nextValFromPostgresSequence = nextValFromPostgresSequence(connection);
+            final long previous = getPrevious(connection);
+
+            updateNextColumnInPreviousRow(nextValFromPostgresSequence, connection);
+            insertRow(eventId, nextValFromPostgresSequence, previous, connection);
+
+            return nextValFromPostgresSequence;
+        }
+    }
+
+    private long nextValFromPostgresSequence(final Connection connection) throws SQLException {
+
+        try (final PreparedStatement preparedStatement = connection.prepareStatement(NEXT_VAL_FROM_SEQUENCE_SQL)) {
+            try (final ResultSet resultSet = preparedStatement.executeQuery()) {
+                resultSet.next();
+                return resultSet.getLong(1);
+            }
+        }
+    }
+
+    private long getPrevious(final Connection connection) throws SQLException {
+
+        try(final PreparedStatement preparedStatement = connection.prepareStatement(SELECT_PREVIOUS_SEQUENCE_NUMBER_SQL)) {
+            try(final ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    return resultSet.getLong(1);
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    private void updateNextColumnInPreviousRow(final long nextValFromPostgresSequence, final Connection connection) throws SQLException {
+
+
+        try(final PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_NEXT_COLUMN_IN_PREVIOUS_ROW_SQL)) {
+
+            preparedStatement.setLong(1, nextValFromPostgresSequence);
+            preparedStatement.executeUpdate();
+        }
+    }
+
+    private void insertRow(final UUID eventId, final long nextValFromPostgresSequence, final long previous, final Connection connection) throws SQLException {
+
+
+        try(final PreparedStatement preparedStatement = connection.prepareStatement(INSERT_NEW_ROW_SQL)) {
+
+            preparedStatement.setObject(1, eventId);
+            preparedStatement.setLong(2, nextValFromPostgresSequence);
+            preparedStatement.setLong(3, previous);
+            preparedStatement.setLong(4, 0);
```

Comment:
Could make it clear why zero is set here...maybe a constant would make the intent clear